google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketPanel.java

```diff
@@ -0,0 +1,158 @@
+    bucketList.setCellRenderer(new GcsBucketCellRenderer());
+    bucketList.setFixedCellHeight(25);
+    bucketList.setBackground(bucketListPanel.getBackground());
+
+    projectSelector
+        .getDocument()
+        .addDocumentListener(
+            new DocumentAdapter() {
+              @Override
+              protected void textChanged(DocumentEvent event) {
+                refresh();
+              }
+            });
+  }
+
+  void refresh() {
+    showNotificationPanel(true);
+
+    if (StringUtils.isEmpty(projectSelector.getText())) {
+      notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.no.project.selected"));
+    } else {
+      String projectId = projectSelector.getText();
+      CredentialedUser user = projectSelector.getSelectedUser();
+
+      if (user != null) {
+        loadAndDisplayBuckets(projectId, user.getCredential());
+      } else {
+        notificationLabel.setText(
+            GctBundle.message("gcs.panel.bucket.listing.error.loading.buckets"));
+      }
+    }
+  }
+
+  @NotNull
+  JPanel getComponent() {
+    return gcsBucketPanel;
+  }
+
+  @VisibleForTesting
+  ProjectSelector getProjectSelector() {
+    return projectSelector;
+  }
+
+  @VisibleForTesting
+  JPanel getNotificationPanel() {
+    return notificationPanel;
+  }
+
+  @VisibleForTesting
+  JPanel getBucketListPanel() {
+    return bucketListPanel;
+  }
+
+  @VisibleForTesting
+  JLabel getNotificationLabel() {
+    return notificationLabel;
+  }
+
+  private void loadAndDisplayBuckets(String projectId, Credential credential) {
+    if (bucketLoadExecution != null && !bucketLoadExecution.isDone()) {
+      return;
+    }
+
+    bucketListModel.clear();
+    notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.loading.text"));
+
+    bucketLoadExecution =
+        ApplicationManager.getApplication()
+            .executeOnPooledThread(
+                () -> {
+                  Storage storage =
+                      GoogleApiClientFactory.getInstance()
+                          .getCloudStorageApiClient(projectId, credential);
+
+                  try {
+                    Iterable<Bucket> buckets = storage.list().iterateAll();
+
+                    if (Iterators.size(buckets.iterator()) == 0) {
+                      notificationLabel.setText(
+                          GctBundle.message("gcs.panel.bucket.listing.no.buckets.found"));
+                      return;
+                    }
+
+                    for (Bucket bucket : buckets) {
+                      bucketListModel.addElement(bucket);
+                    }
+
+                    showNotificationPanel(false);
+                  } catch (StorageException se) {
+                    notificationLabel.setText(
+                        GctBundle.message("gcs.panel.bucket.listing.error.loading.buckets"));
+                  }
+                });
+  }
+
+  private void showNotificationPanel(boolean show) {
```

Comment:
You should avoid boolean params if possible. Consider using two uniquely named methods instead, like `showNotificationPanel()` and `hideNotificationPanel()`

go/java-practices/boolean-parameters